common/djangoapps/entitlements/api/v1/tests/test_views.py

```diff
@@ -124,6 +137,172 @@ def test_add_entitlement(self):
+        that policy is assigned to the entitlement.
+        """
+        policy = CourseEntitlementPolicy.objects.create(mode=CourseMode.PROFESSIONAL, site=None)
+        course_uuid = uuid.uuid4()
+        entitlement_data = self._get_data_set(self.user, str(course_uuid))
+        entitlement_data['mode'] = CourseMode.PROFESSIONAL
+
+        self.client.post(
+            self.entitlements_list_url,
+            data=json.dumps(entitlement_data),
+            content_type='application/json',
+        )
+
+        course_entitlement = CourseEntitlement.objects.get(
+            user=self.user,
+            course_uuid=course_uuid
+        )
+        assert course_entitlement.policy == policy
+
+    def test_set_custom_site_policy_on_create(self):
+        """
+        Verify that, when there does not exist a course entitlement policy with the same mode and site as
+        a created entitlement, but there does exist a policy with the same site and a null mode,
+        that policy is assigned to the entitlement.
+        """
+        course_uuid = uuid.uuid4()
+        entitlement_data = self._get_data_set(self.user, str(course_uuid))
+
+        self.client.post(
+            self.entitlements_list_url,
+            data=json.dumps(entitlement_data),
+            content_type='application/json',
+        )
+        course_entitlement = CourseEntitlement.objects.get(
+            user=self.user,
+            course_uuid=course_uuid
+        )
+
+        policy_site = SiteFactory.create()
+        policy = CourseEntitlementPolicy.objects.create(mode=None, site=policy_site)
+
+        set_entitlement_policy(course_entitlement, policy_site)
+        assert course_entitlement.policy == policy
+
+    def test_set_policy_match_site_over_mode(self):
+        """
+        Verify that, when both a mode-agnostic policy matching the site of a created entitlement and a site-agnostic
+        policy matching the mode of a created entitlement exist but no policy matching both the site and mode of the
+        created entitlement exists, the site-specific (mode-agnostic) policy matching the entitlement is selected over
+        the mode-specific (site-agnostic) policy.
+        """
+        course_uuid = uuid.uuid4()
+        entitlement_data = self._get_data_set(self.user, str(course_uuid))
+
+        self.client.post(
+            self.entitlements_list_url,
+            data=json.dumps(entitlement_data),
+            content_type='application/json',
+        )
+        course_entitlement = CourseEntitlement.objects.get(
+            user=self.user,
+            course_uuid=course_uuid
+        )
+
+        policy_site = SiteFactory.create()
+        policy = CourseEntitlementPolicy.objects.create(mode=None, site=policy_site)
+        CourseEntitlementPolicy.objects.create(mode=entitlement_data['mode'], site=None)
+
+        set_entitlement_policy(course_entitlement, policy_site)
```

Comment:
Same question as above,  You call the POST here which I would assume would set the entitlement policy, but the policy was not created until 250, and then call set_entitlement policy here.

Reply:
The POST also will actually just be setting the policy to None in both of these tests, thus defaulting its policy functionality to the CourseEntitlementPolicy model's default behavior. Since I create the entry into the policy table after creating the entitlement object there's no chance for the policy to be set before I call `set_entitlement_policy` again in the test outside of the context of the POST method itself, but also I could create it beforehand and it would not be assigned to the entitlement due to the site mocking issues I mentioned.